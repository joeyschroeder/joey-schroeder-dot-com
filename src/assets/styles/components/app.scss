$component-selector: app;

.#{$component-selector} {
    height: 100%;
    left: 0;
    position: fixed;
    top: 0;
    width: 100%;
}
