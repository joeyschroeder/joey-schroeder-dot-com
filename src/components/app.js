import React, { Component } from 'react';

import { Sidebar } from './sidebar';
import { Splash } from './splash';

export class App extends Component {

    constructor(props) {
        super(props);
        this.state = {
            sidebarActive: false
        };

        this.toggleSidebar = this.toggleSidebar.bind(this);
    }

    componentDidMount() {
        const event = new CustomEvent('appReady', { bubbles: true, cancelable: false });
        document.dispatchEvent(event);
    }

    toggleSidebar() {
        const { sidebarActive } = this.state;
        this.setState({ sidebarActive: !sidebarActive });
    }

    render() {
        const { sidebarActive } = this.state;

        return (
            <div className={`app ${sidebarActive ? 'app--sidebar-active' : ''}`}>
                <Splash toggleSidebar={this.toggleSidebar} />
                <Sidebar />
            </div>
        );
    }
}
