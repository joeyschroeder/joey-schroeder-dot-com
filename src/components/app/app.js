import 'normalize.css/normalize.css';
import '../../styles/index.scss';

import React, { Component } from 'react';

import { Navigation } from '../navigation/navigation.component';
import { Splash } from '../splash/splash.component';
import styles from './app.style.scss';

export class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      navigationActive: false,
    };

    this.toggleNavigation = this.toggleNavigation.bind(this);
  }

  componentDidMount() {
    // When the app.js component mounts it fires an event called 'appReady'
    const event = new CustomEvent('appReady', {
      bubbles: true,
      cancelable: false,
    });

    document.dispatchEvent(event);
  }

  toggleNavigation() {
    const { navigationActive } = this.state;
    this.setState({ navigationActive: !navigationActive });
  }

  render() {
    const { navigationActive } = this.state;

    return (
      <div className={styles.root}>
        <Splash
          toggleNavigation={this.toggleNavigation}
          active={!navigationActive}
        />
        <Navigation active={navigationActive} />
      </div>
    );
  }
}
