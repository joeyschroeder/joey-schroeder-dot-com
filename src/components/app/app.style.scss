@import "../../styles/config/variables";

.root {
  background: $color-blackType;
  display: flex;
  flex-direction: column;
  height: 100%;
  left: 0;
  overflow: hidden;
  position: fixed;
  top: 0;
  width: 100%;
}
