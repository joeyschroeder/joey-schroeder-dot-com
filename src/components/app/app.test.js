import React from 'react';
import { create } from 'react-test-renderer';
import { App } from './app';

describe('<App />', () => {
  it('should render correctly', () => {
    const component = create(<App />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
