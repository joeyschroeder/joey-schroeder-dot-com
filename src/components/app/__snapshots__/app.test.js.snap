// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should render correctly 1`] = `
<div
  className="root"
>
  <div
    className="root"
    style={
      Object {
        "transform": "translateX(0)",
      }
    }
  >
    <div
      className="root"
      onClick={[Function]}
      style={
        Object {
          "opacity": 0,
          "pointerEvents": "none",
        }
      }
    />
    <button
      className="root"
      onClick={[Function]}
      type="button"
    >
      <span />
      <span />
      <span />
      <span />
      <span />
      <span />
    </button>
    <div
      className="section"
    >
      <div>
        <h1
          className="root"
        >
          Joey Schroeder
        </h1>
        <h2
          className="root"
        >
          <span
            className="content"
          >
            Developer//
          </span>
          <span
            className="content"
          >
            Designer
          </span>
        </h2>
      </div>
    </div>
    <div
      className="section"
    >
      <div
        className="root"
      >
        <div
          className="root"
        />
        <div
          className="root"
        />
        <div
          className="root"
        />
        <div
          className="root"
        />
        <div
          className="root"
        />
      </div>
      <div
        className="root"
      >
        Copyright © 
        2020
        . 
        All Rights Totally Reserved.
      </div>
    </div>
  </div>
  <div
    className="root"
  >
    <a
      className="root"
      download="schroeder-joey-developer-designer-resume.pdf"
      href="test-file-stub"
      rel="noopener noreferrer"
      style={
        Object {
          "transform": "translateX(100%)",
        }
      }
      target="_blank"
      title="Résumé"
    >
      <div
        style={
          Object {
            "marginRight": "1rem",
          }
        }
      >
        <i
          aria-hidden="true"
          className="fas fa-file-pdf"
        />
      </div>
      <div
        style={
          Object {
            "flex": 1,
          }
        }
      >
        Résumé
      </div>
    </a>
    <a
      className="root"
      href="https://www.linkedin.com/in/joeyschroeder"
      rel="noopener noreferrer"
      style={
        Object {
          "transform": "translateX(100%)",
        }
      }
      target="_blank"
      title="LinkedIn"
    >
      <div
        style={
          Object {
            "marginRight": "1rem",
          }
        }
      >
        <i
          aria-hidden="true"
          className="fab fa-linkedin"
        />
      </div>
      <div
        style={
          Object {
            "flex": 1,
          }
        }
      >
        LinkedIn
      </div>
    </a>
    <a
      className="root"
      href="https://github.com/joeyschroeder"
      rel="noopener noreferrer"
      style={
        Object {
          "transform": "translateX(100%)",
        }
      }
      target="_blank"
      title="GitHub"
    >
      <div
        style={
          Object {
            "marginRight": "1rem",
          }
        }
      >
        <i
          aria-hidden="true"
          className="fab fa-github"
        />
      </div>
      <div
        style={
          Object {
            "flex": 1,
          }
        }
      >
        GitHub
      </div>
    </a>
    <a
      className="root"
      href="https://www.npmjs.com/~joeyschroeder"
      rel="noopener noreferrer"
      style={
        Object {
          "transform": "translateX(100%)",
        }
      }
      target="_blank"
      title="NPM"
    >
      <div
        style={
          Object {
            "marginRight": "1rem",
          }
        }
      >
        <i
          aria-hidden="true"
          className="fab fa-npm"
        />
      </div>
      <div
        style={
          Object {
            "flex": 1,
          }
        }
      >
        NPM
      </div>
    </a>
    <a
      className="root"
      href="https://joeyschroeder.bandcamp.com"
      rel="noopener noreferrer"
      style={
        Object {
          "transform": "translateX(100%)",
        }
      }
      target="_blank"
      title="Bandcamp"
    >
      <div
        style={
          Object {
            "marginRight": "1rem",
          }
        }
      >
        <i
          aria-hidden="true"
          className="fab fa-bandcamp"
        />
      </div>
      <div
        style={
          Object {
            "flex": 1,
          }
        }
      >
        Bandcamp
      </div>
    </a>
    <a
      className="root"
      href="https://theapacherevolver.bandcamp.com"
      rel="noopener noreferrer"
      style={
        Object {
          "transform": "translateX(100%)",
        }
      }
      target="_blank"
      title="The Apache Revolver"
    >
      <div
        style={
          Object {
            "marginRight": "1rem",
          }
        }
      >
        <i
          aria-hidden="true"
          className="fab fa-bandcamp"
        />
      </div>
      <div
        style={
          Object {
            "flex": 1,
          }
        }
      >
        The Apache Revolver
      </div>
    </a>
    <a
      className="root"
      href="https://computerjones.bandcamp.com"
      rel="noopener noreferrer"
      style={
        Object {
          "transform": "translateX(100%)",
        }
      }
      target="_blank"
      title="Computer Jones"
    >
      <div
        style={
          Object {
            "marginRight": "1rem",
          }
        }
      >
        <i
          aria-hidden="true"
          className="fab fa-bandcamp"
        />
      </div>
      <div
        style={
          Object {
            "flex": 1,
          }
        }
      >
        Computer Jones
      </div>
    </a>
  </div>
</div>
`;
